### ISSUE TITLE
Issue decoding JSON strings created from Javascript

### ISSUE THREAD

**Issue by darronschall** [mikechambers/as3corelib#18]
Originally filed by dinny.mathew on 2007-02-22T16:29:41

Trying to deserialize following JSON string generated from javscript but
getting parsing error.
[{BookTitle:"JSON", Name:"Mat"}]

But folloing JSON string can be serialized successfully:
[{"BookTitle":"JSON", "Name":"Mat"}]

**Comment by darronschall** [comment]
Updated by mikechambers on 2007-03-03T21:08:09

This may have been fixed by

http://code.google.com/p/as3corelib/issues/detail?id=2&can=1&q=

which is going into the .9 build.

Dinny, can you pull the source and see if that fixes the issue?

**Comment by darronschall** [comment]
Updated by darron.schall on 2007-03-20T23:02:57

This isn't a bug -- the original JSON string that you're trying to use -
[{BookTitle:"JSON", Name:"Mat"}] - is not correctly formatted JSON code.

From json.org, an Object consists of pairs, where each pair is string:value.  The
identifier on the left of the pair then, being a string, needs to be enclosed in quotes.

Whatever library your using to generate the JSON string is not producing
JSON-compliant output.  Make sure that the name part of the name/value pairs inside
of an object is getting correctly wrapped in quotes.

Original ticket set status to Invalid (we converted to closed)
